### PR TITLE
Added support for sync CLI

### PR DESCRIPTION
**Pull request by prklm10**
- sync implementation
- Timeout of 10 min (Not for unit test)

### CODE REVIEW

File: Percy/Percy.cs [hunk 1]
```diff
@@ -139,11 +148,12 @@ private static string GetPercyDOM()
 
         public class Options : Dictionary<string, object> {}
 
-        public static void Snapshot(
+        public static JObject Snapshot(
             WebDriver driver, string name,
-            IEnumerable<KeyValuePair<string, object>>? options = null)
+            IEnumerable<KeyValuePair<string, object>>? options = null,
+            Boolean sync = false)
```
Comment (chinmay-browserstack) [inline Percy/Percy.cs:154]: sync should be inside options as other parameter

File: Percy/Percy.cs [hunk 2]
```diff
@@ -67,6 +72,10 @@ private static dynamic Request(string endpoint, object? payload = null, bool isJ
         {
             StringContent? body = payload == null ? null : new StringContent(
                 PayloadParser(payload, isJson), Encoding.UTF8, "application/json");
+            if (!IsRunningFromXUnit) {
+                _http = new HttpClient();
+                _http.Timeout = TimeSpan.FromMinutes(10);
+            }
```
Comment (chinmay-browserstack) [inline Percy/Percy.cs:78]: why?

File: Percy/Percy.cs [hunk 3]
```diff
@@ -172,25 +183,31 @@ public class Options : Dictionary<string, object> {}
...
             catch(Exception error)
             {
                 Log($"Could not take DOM snapshot \"{name}\"");
                 Log(error);
+                return null;
             }
         }
 
-        public static void Screenshot(WebDriver driver, string name, IEnumerable<KeyValuePair<string, object>>? options = null)
+        public static JObject Screenshot(WebDriver driver, string name, IEnumerable<KeyValuePair<string, object>>? options = null)
```
Comment (chinmay-browserstack) [inline Percy/Percy.cs:199]: return type should void is sync is not passed

File: Percy/Percy.cs [hunk 4]
```diff
@@ -245,38 +262,48 @@ public static void Screenshot(WebDriver driver, string name, IEnumerable<KeyValu
                             userOptions["consider_region_elements"] = elementIds;
                         }
                     }
+                    userOptions[SYNC] = isSync;
```
Comment (chinmay-browserstack) [inline Percy/Percy.cs:265]: isn't this will set value of sync false even when true is passed by user?

File: Percy/Percy.cs [hunk 5]
```diff
@@ -245,38 +262,48 @@ public static void Screenshot(WebDriver driver, string name, IEnumerable<KeyValu
...
             catch(Exception error)
             {
                 Log($"Could not take Percy Screenshot \"{name}\"");
                 Log(error);
+                return null;
             }
         }
 
-        public static void Snapshot(WebDriver driver, string name, object opts)
+        public static JObject Snapshot(WebDriver driver, string name, object opts, bool sync = false)
```
Comment (chinmay-browserstack) [inline Percy/Percy.cs:290]: move sync inside options + no default value for sync

File: Percy/packages.lock.json [hunk 6]
```diff
@@ -19,32 +19,32 @@
       },
       "Selenium.WebDriver": {
         "type": "Direct",
-        "requested": "[4.0.0, )",
-        "resolved": "4.0.0",
-        "contentHash": "9v3Dfq/y2j0QMNu0X6YsnKTPstR2rhgbmO9yV/WWoQFBAmPN8yLHoSX6S9gdDUjQMfYJ9gaNFCHX2ZpUo4mQQA==",
+        "requested": "[4.5.1, )",
+        "resolved": "4.5.1",
+        "contentHash": "ztu9Qv2n5/NkqPs0MQ3nojMJuW2PVHrAKDukmT77HbYAGhZI370Hp0EJNqyucJuud937Lr/qvXsoOtLZ18VFnw==",
```
Comment (chinmay-browserstack) [inline Percy/packages.lock.json:24]: ?

File: Percy/Percy.cs (hunk 2 above)
Comment (prklm10) [reply Percy/Percy.cs:78]: tests are failing because we are assigning new _http instance

File: Percy/packages.lock.json (hunk 6 above)
Comment (prklm10) [reply Percy/packages.lock.json]: reverted

File: Percy/Percy.cs (hunk 5 above)
Comment (prklm10) [reply Percy/Percy.cs]: done

File: Percy/Percy.cs (hunk 4 above)
Comment (prklm10) [reply Percy/Percy.cs]: removed

File: Percy/Percy.cs (hunk 1 above)
Comment (prklm10) [reply Percy/Percy.cs]: done

File: Percy/Percy.cs [hunk 7]
```diff
@@ -160,7 +170,8 @@ public class Options : Dictionary<string, object> {}
                     { "environmentInfo", ENVIRONMENT_INFO },
                     { "domSnapshot", domSnapshot },
                     { "url", driver.Url },
-                    { "name", name }
+                    { "name", name },
+                    { "sync", sync }
```
[the remark this answers is not in the document]
Comment (prklm10) [reply Percy/Percy.cs]: test passed

File: Percy/Percy.cs (hunk 2 above)
Comment (chinmay-browserstack) [reply Percy/Percy.cs:78]: we should fix testcase in that case.

File: Percy/Percy.cs [hunk 8]
```diff
@@ -139,11 +150,11 @@ private static string GetPercyDOM()
 
         public class Options : Dictionary<string, object> {}
 
-        public static void Snapshot(
+        public static JObject Snapshot(
```
Comment (ninadbstack) [inline Percy/Percy.cs:153]: should be JObject?

**Review by Amit3200 — APPROVED** [review]

**Pull request merged**